students/ScottL/session04/Mailroom_Part2.py

```diff
@@ -0,0 +1,185 @@
+            sorted_donation_list.append({"Name": item["Name"], "Amount": item["Amount"]})
+    sorted_donation_list.sort(key=lambda x: x["Amount"], reverse=True)
+    return sorted_donation_list
+
+
+def count_num_of_donations_per_donor():
+    """Return a dictionary of donors and their respective number of donations."""
+    number_of_donations = {}
+    for item in global_donation_list:
+        if item["Name"] in number_of_donations:
+            number_of_donations[item["Name"]] += 1
+        else:
+            number_of_donations[item["Name"]] = 1
+    return number_of_donations
+
+
+def calc_avg_donation_per_donor(list_of_donations, donation_number):
+    """
+    Return a dictionary of the mean donation size for each donor.
+    arg1 = list containing a string of names in the first element
+            and an integer/float in the second element
+    arg2 = list containing a string of names in the first element
+            and an integer in the second element
+    """
+    average_donation_dict = {}
+    for item in list_of_donations:
+        average_donation = item["Amount"] / donation_number[item["Name"]]
+        average_donation_dict[item["Name"]] = average_donation
+    return average_donation_dict
+
+
+def display_donor_report(donors, donations_count, average_donations):
+    """
+    Display the donors names, total donations, number of donations, and mean donation size.
+    arg1 = a list containing names in the first element and the total donation in the second
+    arg2 = a dictionary with donor names as the key to correspond with donor names in arg1
+    arg3 = a dictionary with donor names as the key to correspond with donor names in arg1
+    """
+    print(REPORT_HEADER_1, REPORT_HEADER_2)
+    for item in donors:
+        print("{:<26}${:>12}{:^17}${:>11}".format(item["Name"], item["Amount"],
+                                                  donations_count[item["Name"]],
+                                                  round(average_donations[item["Name"]])))
+
+
+def display_donor_names():
+    """Print the list of full donor names for the user."""
+    print("\n", DONOR_LIST_HEADER)
+    donor_list = calc_total_donations_per_donor()
+    for entry in donor_list:
+        print(entry["Name"])
+
+
+def send_email(name, amount):
+    """
+    Print a statement thanking the donor for their new donation.
+    arg1 = a string containing the donor's name
+    arg2 = a number containing the amount of the donation
+    """
+    print("\nDear {},\nThanks for the generous donation of ${}."
+          "\n\nSincerely,\nDonations Staff".format(name, amount))
+
+
+def add_donation(donor_name):
+    """
+    Return an appended donation list.
+    arg1 = a string containing the donor's name
+    """
+    donation_amount = input("Please enter the donation amount (or enter 'Quit' to return to "
+                            "Main Menu): ").strip("$")
+    if donation_amount.strip().lower() == "quit":
+        pass
```

Comment:
if your not doing anything with the value "quit" there is no reason to check for it.  Same on line 131

Reply:
it allows the user to abort the input.  the instructions said to allow for the user to enter quit at any point.  it was structured this way to easily skip over completing the task if 'quit' was entered.

Reply:
You'd want to do something like this then
```
donation_amount = input("Please enter the donation amount (or enter 'Quit' to return to "
                            "Main Menu): ").strip("$")
    if donation_amount.strip().lower() != "quit":
         global_donation_list.append({"Name": donor_name.title(), "Amount":  int(donation_amount)})
         send_email(donor_name.title(), donation_amount)

Reply:
Yep, that makes a lot of sense.  Appreciate the help